.github/workflows/release.yml

```diff
@@ -9,118 +9,97 @@ on:
       - 'client-v*'
   workflow_dispatch: # Allow manual creation of artifacts without a release
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
   build:
     name: build release assets
-    runs-on: ${{ matrix.config.os }}
+    runs-on: ${{ matrix.config.runnerOs }}
     outputs:
       version_output: ${{ steps.version_output.outputs.version }}
     strategy:
       matrix:
         config:
-          # NOTE: We are building on an older version of ubuntu because of libc compatibility
-          # issues. Namely, if we build on a new version of libc, it isn't backwards compatible with
-          # old versions. But if we build on the old version, it is compatible with the newer
-          # versions running in ubuntu 22 and its ilk
           - {
-              os: 'ubuntu-20.04',
-              arch: 'amd64',
-              extension: '',
-              targetPath: 'target/release/',
+              runnerOs: 'ubuntu-latest',
+              target: 'x86_64-unknown-linux-musl',
+              uploadArtifactSuffix: 'linux-amd64',
+              buildOutputPath: 'target/release/wadm',
             }
           - {
-              os: 'ubuntu-20.04',
-              arch: 'aarch64',
-              extension: '',
-              targetPath: 'target/aarch64-unknown-linux-gnu/release/',
+              runnerOs: 'ubuntu-latest',
+              target: 'aarch64-unknown-linux-musl',
+              uploadArtifactSuffix: 'linux-aarch64',
+              buildOutputPath: 'target/aarch64-unknown-linux-musl/release/wadm',
             }
           - {
-              os: 'macos-13',
-              arch: 'amd64',
-              extension: '',
-              targetPath: 'target/release/',
+              runnerOs: 'macos-14-large',
```

Comment:
Is there a reason this one is using large?

Reply:
I picked these values based on my reading of the [Available Images chart GitHub publishes for their runner images](https://github.com/actions/runner-images?tab=readme-ov-file#available-images), based on my understanding, if you want an arm64-based runner you have to pick `macos-14-large` instead of `macos-14`

Reply:
We could also use `macos-13` for x86 and `macos-14` for arm64, but that felt a bit strange, open to suggestions though!

Reply:
Ended up using the same runner image for both the x86 and arm64 builds.